eng/WpfArcadeSdk/tools/Wpf.Cpp.targets

```diff
@@ -436,6 +436,145 @@ using namespace System::Runtime::Versioning;
+  -->
+  <Target Name="CppHelper"
+          Returns="@(NuGetReferencesForCpp)"
+          Condition="'$(ManagedCxx)' != 'true' and '$(RepoLocation)' == 'External'">
+
+    <PropertyGroup>
+      <CppHelperProject>$(IntermediateOutputPath)CppSupport\CppSupport_$(MSBuildProjectName).csproj</CppHelperProject>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <CppSupportProject Include="ProjectHeader">
+        <!-- 
+            In the CDATA section that follows, 
+              %40 ~= @
+              %25 ~= % 
+              %3B ~= ;
+              %24 ~= $
+          -->
+        <Text>
+          <![CDATA[
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>netcoreapp5.0</TargetFramework>
+    <Platforms>AnyCPU%3Bx64</Platforms>
+  </PropertyGroup>
+  ]]>
+        </Text>
+      </CppSupportProject>
+      
+      <!-- 
+        We need a reference to Microsoft.DotNet.Wpf.GitHub 
+      -->
+      <CppSupportProject Include="MicrosoftDotNetWpfGitHubPackageReference">
+        <Text>
+          <![CDATA[
+   <ItemGroup>
+      <PackageReference Include="runtime.$(WpfRuntimeIdentifier).Microsoft.DotNet.Wpf.DncEng"
+                      Version="$(MicrosoftDotNetWpfDncEngVersion)" 
+                      GeneratePathProperty="true" />
+  </ItemGroup>
+  ]]>
+        </Text>
+      </CppSupportProject>
+      
+      <CppSupportProject Include="ProjectBody">
+        <Text>
+          <![CDATA[
+  <!-- Ensure that LimitMicrosoftDotNetWpfGitHubReferences target doesn't run --> 
+  <PropertyGroup>
+    <DoNotLimitMicrosoftDotNetWpfGitHubReferences>true</DoNotLimitMicrosoftDotNetWpfGitHubReferences>
+  </PropertyGroup>
+  
+<Target
+    Name="IdentifyNuGetReferences"
+    AfterTargets="ResolveAssemblyReferences"
+    Returns="%40(NuGetReferencesForCpp)">
+
+   <PropertyGroup>
+      <WpfDncEngRidSpecificLibPath Condition="'%24(RuntimeIdentifier)'=='win-x86' Or '%24(WpfRuntimeIdentifier)'=='win-x86'">%24(Pkgruntime_win-x86_Microsoft_DotNet_Wpf_DncEng)\runtimes\%24(RuntimeIdentifier)\native\</WpfDncEngRidSpecificLibPath>
+      <WpfDncEngRidSpecificLibPath Condition="'%24(RuntimeIdentifier)'=='win-x64' Or '%24(WpfRuntimeIdentifier)'=='win-x64'">%24(Pkgruntime_win-x64_Microsoft_DotNet_Wpf_DncEng)runtimes\%24(RuntimeIdentifier)\native\</WpfDncEngRidSpecificLibPath>
+   </PropertyGroup>
+<!-- Save Microsoft.NETCore.App assemblies and return them -->
+    <ItemGroup>
+      <NuGetReferencesForCpp Remove="%40(NuGetReferencesForCpp)" />
+      <NuGetReferencesForCpp Include="%24(WpfDncEngRidSpecificLibPath)**\*.*" />
+    </ItemGroup>
+  </Target>
+</Project>
+        ]]>
+        </Text>
+      </CppSupportProject>
+    </ItemGroup>
+
+    <MakeDir Condition="!Exists('$(IntermediateOutputPath)CppSupport\')"
+             Directories="$(IntermediateOutputPath)CppSupport\" />
+
+    <Delete Files="$(CppHelperProject)" Condition="Exists('$(CppHelperProject)')" />
+
+    <WriteLinesToFile Lines="%(CppSupportProject.Text)"
```

Comment:
Maybe we can use `MSBuild Escape` that we can use `;` and `@` etc.

See https://docs.microsoft.com/en-us/visualstudio/msbuild/how-to-escape-special-characters-in-msbuild?view=vs-2019 and https://blog.lindexi.com/post/Roslyn-how-to-use-WriteLinesToFile-to-write-the-semicolons-to-file.html

Here is the example from my blog

```
    <PropertyGroup>
      <SomeThing>
        <![CDATA[
            Console.WriteLine("I am lindexi");
       ]]>
      </SomeThing>
    </PropertyGroup>
    <WriteLinesToFile File="lindexi.cs" Lines="$([MSBuild]::Escape($(SomeThing)))" Overwrite="true"></WriteLinesToFile>
```

Reply:
This is a tested pattern, so I don't want to change this. Most of this pattern will go away shortly when we switch to adopting vcxproj support in Dev16.4 for C++/CLI. What remains will be minimal and easy enough to maintain (and is not intended to be maintained by the community at large- its mainly targeted at a couple of maintainers in the WPF team).